net: openthread: remove unneeded `OPENTHREAD_MAX_CHILDREN` default value

edmont opened the pull request:
After https://github.com/openthread/openthread/pull/9213 there is no need to reduce the `ChildMask` size for MTD builds.

pdgendt reviewed (approved):
Nit: typo "unneded" in commit subject

edmont committed:
net: openthread: remove unneeded `OPENTHREAD_MAX_CHILDREN` default value

After https://github.com/openthread/openthread/pull/9213 there is no
need to reduce the `ChildMask` size for MTD builds.

Signed-off-by: Eduardo Montoya <eduardo.montoya@nordicsemi.no>
